### PR TITLE
Retry attempt if specific error message is found

### PR DESCRIPTION
**Pull request by cloudmagic80**
## Summary
- Added retry logic for file uploads when a specific error is detected.
- If the `statuses` array contains a "No such file or directory @ rb_sysopen" error message, the system re-fetches file paths and metadata, and retries the upload process.
- This change improves resilience by automatically handling temporary file path issues during the upload process.
- This PR will be merged after this PR goes in.
- https://github.com/department-of-veterans-affairs/vets-api/pull/19236
- 

Story:
https://app.zenhub.com/workspaces/ivc-forms-652da2d3f0ae4c0016bfb109/issues/gh/department-of-veterans-affairs/va.gov-team/95561

## Testing done

Tested on F1 and 1010D. 
Checked tmp folder
Ran unit test
Checked s3 bucket

## Screenshots
UUID 4f280fc2-c107-4d29-a7eb-710ec87da252_vha_10_7959f_1-tmp got uploaded.
UUID 510bd938-e276-46ac-b2d1-fc5610fe4240_vha_10_7959f_1-tmp111111 was the pdf that is missing. I renamed so that it couldn't be found.
Screenshot 2024-10-22 at 9 50 51 AM

Screenshot 2024-10-22 at 9 52 01 AM

## What areas of the site does it impact?
Only the IVC Champva uploads to s3

## Acceptance criteria
Upload the pdf into the Pega S3 bucket

### CODE REVIEW

**Review by reiting — DISMISSED** [review]

**Comment by rmtolmach** [comment]
You were 142 commits behind master 😳 so I've updated your branch.

**Review by rmtolmach — DISMISSED** [review]

File: modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb [hunk 1]
```diff
@@ -15,26 +15,49 @@ class UploadsController < ApplicationController
...
 
-      def submit
-        Datadog::Tracing.trace('Start IVC File Submission') do
-          form_id = get_form_id
-          Datadog::Tracing.active_trace&.set_tag('form_id', form_id)
-          parsed_form_data = JSON.parse(params.to_json)
-          file_paths, metadata = get_file_paths_and_metadata(parsed_form_data)
-          statuses, error_message = FileUploader.new(form_id, metadata, file_paths, true).handle_uploads
-          response = build_json(Array(statuses), error_message)
+      if Flipper.enabled?(:champva_file_recreate, @user)
```
Comment (stevelong00) [inline modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb:18]: This feature toggle needs to be defined in `config/features.yml`

Comment (stevelong00) [reply modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb:18]: Looks like that will happen once [this PR](https://github.com/department-of-veterans-affairs/vets-api/pull/19236) is merged

**Review by stevelong00 — COMMENTED** [review]
Everything looks good and tests good.  I will approve once [this blocking PR](https://github.com/department-of-veterans-affairs/vets-api/pull/19236) is merged.

File: modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb (hunk 1 above)
Comment (cloudmagic80) [reply modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb:18]: Yup. It is defined in config/features.yml. Search for "champva_file_recreate"

**Review by stevelong00 — APPROVED** [review]
Looks good!

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**